### DISCUSSION TITLE
Default resolv conf

### DISCUSSION THREAD

**Discussion by vassilv** [sakai135/wsl-vpnkit#218] [General]
Hi Sakai,

After days of fight, trying to make my WSL has internet connection behind VPN, I finally found this tool. It took me some time to configure it correctly but at the end it works and now I have internet on VPN. You've done amazing job.

I'm pretty new to WSL and maybe it's a dumb question, and I'm missing something but let me ask you. When I was playing and trying to make it work I read a lot of articles how to configure the resolv and wsl config files. So I updated them and now the file /etc/resolv.conf is properly configured and immutable. But it took me some time to figure out (I should have read the docs first :)) that your script first reads the default resovl conf from /mnt/wsl/resolv.conf and if it's missing - from /etc/resolv.conf. Once I updated /mnt/wsl/resolv.conf, the internet connection was fixed. The problem I have now is that this file /mnt/wsl/resolv.conf is autogerated every time WSL is restarted, which is not the case with /etc/resolv.conf.

**Comment by sakai135** [comment]
Thanks for the post, and sorry for the late reply. The resolv.conf was used to get the IP address of the WSL2 machine from inside the WSL2 VM. The recommended way to get around it is to specify it with environment variables to the script.
